### ISSUE TITLE
🐞 dagger doesn't work with sysbox

### ISSUE THREAD

**Issue by vbehar** [dagger/dagger#6255]
### What is the issue?

I've tried to run Dagger in docker-in-docker on top of sysbox, but it doesn't work - because it seems like sysbox doesn't allow Dagger to create a new network namespace

### Log output

time="2023-12-12T09:19:13Z" level=info msg="detected mtu 1500 via interface eth0"
dnsmasq[89]: started, version 2.89 cachesize 150
dnsmasq[89]: compile time options: IPv6 GNU-getopt no-DBus no-UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP no-conntrack ipset no-nftset auth no-cryptohash no-DNSSEC loop-detect inotify dumpfile
dnsmasq[89]: warning: interface dagger0 does not currently exist
dnsmasq[89]: using only locally-known addresses for dagger.local
time="2023-12-12T09:19:13Z" level=debug msg="setting up engine tracing"
dnsmasq[89]: reading /etc/dnsmasq-resolv.conf
dnsmasq[89]: using nameserver 172.20.0.10#53
dnsmasq[89]: using only locally-known addresses for dagger.local
time="2023-12-12T09:19:13Z" level=error msg="failed to create tracer provider" error="cannot merge resource due to conflicting Schema URL"
dnsmasq[89]: read /var/run/containers/cni/dnsname/dagger/addnhosts - 0 names
time="2023-12-12T09:19:13Z" level=debug msg="creating engine GRPC server"
time="2023-12-12T09:19:13Z" level=debug msg="creating engine lockfile"
time="2023-12-12T09:19:13Z" level=debug msg="creating engine controller"
time="2023-12-12T09:19:13Z" level=error msg="failed to create tracer exporter" error="cannot merge resource due to conflicting Schema URL"
time="2023-12-12T09:19:13Z" level=info msg="auto snapshotter: using overlayfs"
time="2023-12-12T09:19:13Z" level=debug msg="could not read \"/var/lib/dagger/net/cni\" for cleanup: open /var/lib/dagger/net/cni: no such file or directory"
time="2023-12-12T09:19:13Z" level=debug msg="creating new network namespace yk686y8k59878nrcavwq2cto2"
buildkitd: permission denied

### Steps to reproduce

- create the dagger-engine container in debug mode in dind
- see that the container stops soon after being created

### SDK version

0.9.4

### OS version

linux something (sorry I don't know, I don't control it)

**Comment by vbehar** [comment]
note that I had it working with dagger 0.8 by disabling the services feature - so basically not using the CNI

with dagger 0.9 this is not possible.

so a workaround is to keep using dagger 0.8 with `_EXPERIMENTAL_DAGGER_SERVICES_DNS=0`
or to use a dagger engine set up elsewhere...

**Comment by shykes** [comment]
Paging @vito @jedevc @sipsma who may have more insights into the internals.

@vbehar what is your flexibility in changing your setup? Are you using sysbox because there's no other way? Can you share details on your infrastructure? I believe this is a gitlab runner - is it running on kubernetes? Would you have the option of running a sidecar container, to bypass docker-in-docker altogether?

**Comment by vbehar** [comment]
yes, we're using a shared set of gitlab runners hosted in kubernetes - using sysbox to get a "safe" way to allow docker-in-docker.

for our dagger builds, we've just setup an extra docker-based gitlab runner running in a single box, so that all our dagger builds uses the same dagger-engine instance, and we get local caching.
for now it's enough for our scale - we'll see later when we'll need to scale if more projects move to dagger

**Comment by shykes** [comment]
> for our dagger builds, we've just setup an extra docker-based gitlab runner running in a single box, so that all our dagger builds uses the same dagger-engine instance, and we get local caching. for now it's enough for our scale - we'll see later when we'll need to scale if more projects move to dagger

Ok. Just so you know, you can run Dagger as a DaemonSet on your cluster: each node in the cluster gets a dedicated Dagger engine, with persisted local storage (up to the lifetime of the node). Clients in each Gitlab runner pod route their requests to their local node's engine.

Something to keep in mind when you look for an upgrade from your single-node configuration.

**Comment by bgdnlp** [comment]
I got the same error when attempting to get a quick idea about Dagger using AWS Cloud Shell.

I'm just mentioning it because the cause might be the same and it's easier to reproduce there. I don't intend to actually keep using AWS Cloud Shell, but it might be a quick and free way to go through the quickstart and see what Dagger is.

**Comment by NAVRockClimber** [comment]
I really would love to be able to run this unpriveleged in one or another way as well. This is currently the only thing stopping me from using this awesome tool.
